### ISSUE TITLE
Terminating JSON files with newlines

### ISSUE THREAD

**Issue by maroneze** [ocaml-community/yojson#109]
In a few situations, such as for version control with strict checking, having JSON files with no newlines at the end of the file can be bothersome, and is apparently un-POSIX (people have reported similar issues in other tools: https://github.com/netlify/netlify-cms/issues/1382, https://github.com/serverless/serverless/issues/1321)

I wonder if you have already considered the possibility of adding a newline when printing JSON objects to files, e.g. via `Yojson.to_file`, or if there are specific reasons not to do it, and that it should be done by the yojson user via other functions.

**Comment by mjambon** [comment]
The original idea was that the `to_file` function would be essentially the same as `to_channel` or `to_buffer` without surprises. Those last two functions don't print a newline because that's often what we want. So, `to_file` simply mirrors the behavior of `to_channel` or `to_buffer` and doesn't print a newline either.

It seems reasonable to at least provide an option to print a newline at the end of the object. Perhaps a signature like this would be nice:
```ocaml
val to_file :
  ?len:int ->
  ?newline:bool ->
  ?std:bool ->
  string -> t -> unit
```
with the default value for `newline` being `false`, for backward compatibility. If backward compatibility is no big issue, I'd probably make the default value `true`. That's just a quick suggestion. I'll let the new maintainers decide.

**Comment by maroneze** [comment]
Yes, I think the surprising behavior might be inconvenient, so the optional argument would be nice. Otherwise, just a simple mention of this fact in the documentation would also be fine for me.

**Comment by panglesd** [comment]
Closed by #124.
